src/contracts/ODSaviour.sol

```diff
@@ -55,14 +56,36 @@ contract ODSaviour is Authorizable, Modifiable, ModifiablePerCollateral, IODSavi
     safeEngine = ISAFEEngine(address(safeManager.safeEngine()));
   }
 
-  function isEnabled(uint256 _vaultId) external view returns (bool _enabled) {
+  function isEnabled(uint256 _vaultId) public view returns (bool _enabled) {
     _enabled = _enabledVaults[_vaultId];
   }
 
   function cType(bytes32 _cType) public view returns (address _tokenAddress) {
     return address(_saviourTokenAddresses[_cType]);
   }
 
+  function saviourIsReady(bytes32 _cType) public view returns (bool) {
+    return (IERC20(_saviourTokenAddresses[_cType]).allowance(saviourTreasury, address(this)) != 0)
+      && (ILiquidationEngine(liquidationEngine).safeSaviours(address(this)) != 0);
+  }
+
+  function vaultSafteyCheck(uint256 vaultId) public view returns (VaultSaftey memory health) {
+    health.vaultId = vaultId;
+    IODSafeManager.SAFEData memory safeData = safeManager.safeData(vaultId);
+    health.allowed = safeManager.safeCan(safeData.owner, vaultId, safeData.nonce, address(this));
+    health.enabled = isEnabled(vaultId);
+    health.vaultCtypeTokenAddress = cType(safeData.collateralType);
+    if (health.vaultCtypeTokenAddress != address(0)) {
+      health.saviourAllowance = IERC20(health.vaultCtypeTokenAddress).allowance(saviourTreasury, address(this));
+    } else {
+      revert UninitializedCollateral(safeData.collateralType);
+    }
+    health.safeProtected = ILiquidationEngine(liquidationEngine).chosenSAFESaviour(
+      safeData.collateralType, safeData.safeHandler
+    ) == address(this);
+    health.saviourIsReady = health.allowed && health.enabled && (health.saviourAllowance != 0) && health.safeProtected;
```

Comment:
The naming `saviourIsReady` conflicts with the function by the same name. I think we should just remove this one.